### PR TITLE
feat: Add Custom Slack Heading

### LINKED ISSUE #54: Customizable "Title" Message
## Feature Request

A deployment may require more context than a static message that it has been deployed. A build number, an environment, or even whether the changelog indicates that something has actually been "deployed", depending on a projects workflow.

## Possible Solutions
In increasing complexity:

- An optional, static title defined in yaml
- An optional, scripted title that can be executed at build time (using a Git tag or file contents as input, for example)

(I may submit a PR, if I overcome my lifelong avoidance of JS 😄)

### PR DESCRIPTION
**Pull request by therynamo**
closes #54 

This adds a single field to the configuration called `slack_message_heading`. You can use this field to populate the Captain's Log heading with whatever you would like. 

Note: when #55 is merged, we'll need to update this pull request or have a subsequent pull request that will add this field to the external config.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -11,7 +11,7 @@ const {
 const defaultTeam = Team();
 const teamList = teams.length ? teams.map(team => Team(team)) : [];
 
-const createAttachment = (hasMessages, { owner, repo }) => {
+const createAttachment = (hasMessages, { owner, repo, config }) => {
```
Comment (AndersDJohnson) [inline src/index.js]: Who is calling `createAttachment` and are they passing in `config`? I see it's not being passed here:

https://github.com/target/captains-log/blob/4dbc061a109d21e7992d0095ed40e7ab5bc06906/src/index.js#L84

File: docs/configuration/README.md [hunk 2]
```diff
@@ -14,11 +14,12 @@
 
 **Optional Fields**
 
-- `jira_team_domain` - **Required** if you use the Jira issue tracker. Namespace of the Jira your workspace (e.g. `jira.myteamnamespace.com`, **myteamnamespace** would be the value for this key).
+- `enterprise_host` - **Required if** you use Enterprise Github. This is where you would supply the custom domain. (e.g. https://git.myCompany.com, `/api/v3` will be appended to this **do not include it for this value**).
```
Comment (AndersDJohnson) [inline docs/configuration/README.md]: This `enterprise_host` was missing in the docs and you meant to add it here?

Comment (therynamo) [reply docs/configuration/README.md]: If you look further down in the diff, I just moved it up.

File: src/index.js (hunk 1 above)
Comment (therynamo) [reply src/index.js]: Huh, I swore that I passed that change. Thanks for catching that 👍

File: src/index.js [hunk 3]
```diff
@@ -21,7 +21,8 @@ const createAttachment = (hasMessages, { owner, repo }) => {
   }
 
   // add all the PRs if there are any
-  message = DEFAULT_HEADING(owner, repo);
+  const customHeading = config.get('slack:messageHeading');
+  message = customHeading || DEFAULT_HEADING(owner, repo);
```
Comment (AndersDJohnson) [inline src/index.js]: A constant function? Feels a little odd, but we can go with it.

File: docs/configuration/README.md [hunk 4]
```diff
@@ -14,11 +14,12 @@
...
 - `github_tag_id` - you can use this as regex to match on specific tags. Useful when your tags are not in chronological order (e.g. `-rc0`, `-rc1`, `-release`).
 - `slack_channel`/`SLACK_URL` - when using the API, you should use `slack_channel` to specify which room you'd like to post to. When using `SLACK_URL` you should **not** specify the room (i.e. `slack_channel`) because the room is already a part of the incoming webhook. ([Setting Up A Webhook (e.g. SLACK_URL)](https://api.slack.com/incoming-webhooks), [Setting Up A Slack Token](https://api.slack.com/docs/token-types#verification))
 - `teams` - a list of teams which allows you to organize the output of Captain's Log into meaningful chunks. (more below)
-- `enterprise_host` - **Required** if you use Enterprise Github. This is where you would supply the custom domain. (e.g. https://git.myCompany.com, `/api/v3` will be appended to this **do not include it for this value**).
+- `slack_message_heading` - you can provide a custom heading for the slack message that is posted, when using this field.
```
Comment (AndersDJohnson) [inline docs/configuration/README.md]: Any desire to align the camel case names of these values in the `nconf` instance with the snake case names in config files and environment variables?

**Pull request merged**